skbio/metadata/tests/test_interval.py

```diff
@@ -0,0 +1,417 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import unittest
+
+from skbio.metadata._interval import _polish_interval
+from skbio.metadata._interval import Interval
+from skbio.metadata import IntervalMetadata
+
+
+class TestInterval(unittest.TestCase):
+
+    def test_constructor(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+        self.assertTrue(f._interval_metadata is not None)
+        self.assertListEqual(f.intervals, [(1, 2), (4, 7)])
+        self.assertListEqual(f.boundaries, [(True, False), (False, False)])
+        self.assertDictEqual(f.metadata, {'name': 'sagA',
+                                          'function': 'transport'})
+
+    def test_bad_constructor(self):
+        with self.assertRaises(ValueError):
+            Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[1, (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+    def test_repr(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        exp1 = (r"Interval(intervals=[(1, 2), (4, 7)], "
+                "metadata={'name': 'sagA', 'function': 'transport'})")
+        res = repr(f)
+        # because dictionaries are random
+        self.assertTrue(res, exp1)
```

Comment:
this should be `assertEqual` or `assertDictEqual`, as is, this test is not working as expected